client/extensions/woocommerce/state/sites/reviews/handlers.js

```diff
@@ -62,3 +79,99 @@ export function handleReviewsRequestError( { dispatch }, action, error ) {
 		error,
 	} );
 }
+
+export function handleChangeReviewStatus( { dispatch, getState }, action ) {
+	const { siteId, reviewId, newStatus } = action;
+	// The product reviews endpoint batch/update endpoints seem busted in v3, v2, and v1
+	// We can use the comments endpoint for now. One difference is pending vs hold
+	const statusToPass = 'pending' === newStatus ? 'hold' : newStatus;
+	dispatch( request( siteId, action, '/wp/v2' ).post( `comments/${ reviewId }`, { status: statusToPass } ) );
+}
+
+export function handleChangeReviewStatusSuccess( { dispatch, getState }, action ) {
+	const state = getState();
+	const { siteId, currentStatus, newStatus } = action;
+
+	// Refreshes the review list if dealing with trash and spam.
+	// otherwise pending/approved reviews stay in their current lists.
+	// This matches the behavior of comments management.
+	if ( 'spam' === currentStatus || 'spam' === newStatus || 'trash' === currentStatus || 'trash' === newStatus ) {
+		const currentPage = getReviewsCurrentPage( state, siteId );
+		const currentSearch = getReviewsCurrentSearch( state, siteId );
+		const query = {
+			page: currentPage,
+			search: currentSearch,
+			status: currentStatus,
+		};
+
+		if ( '' !== currentSearch ) {
+			query.status = 'any';
+		}
+
+		dispatch( fetchReviews( siteId, query ) );
+	}
+
+	const message = {
+		approved: translate( 'Review approved.' ),
+		pending: translate( 'Review unapproved.' ),
+		spam: translate( 'Review marked as spam.' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Comment marked as spam.' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).